projects/kit/components/input-phone/input-phone.component.ts

```diff
@@ -166,56 +156,41 @@ export class TuiInputPhoneComponent
     }
 
     get canClean(): boolean {
-        return this.computedValue !== this.countryCode && this.textfieldCleaner.cleaner;
-    }
-
-    onDrop(event: DragEvent): void {
-        if (!event.dataTransfer) {
-            return;
-        }
-
-        this.setValueWithoutPrefix(event.dataTransfer.getData('text'));
-        event.preventDefault();
+        return (
+            this.nativeValue !== this.nonRemovablePrefix && this.textfieldCleaner.cleaner
+        );
     }
 
-    onPaste(event: Event): void {
-        this.setValueWithoutPrefix(tuiGetClipboardDataText(event as ClipboardEvent));
+    get maskOptions(): MaskitoOptions {
+        return this.calculateMask(
+            this.countryCode,
+            this.phoneMaskAfterCountryCode,
+            this.nonRemovablePrefix,
+            this.allowText,
+        );
     }
 
     onActiveZone(active: boolean): void {
         this.updateFocused(active);
 
-        if (active && !this.computedValue && !this.readOnly && !this.allowText) {
-            this.updateSearch(this.countryCode);
+        if (active && !this.nativeValue && !this.readOnly && !this.allowText) {
+            this.updateSearch(this.nonRemovablePrefix);
+            this.nativeValue = this.nonRemovablePrefix;
 
             return;
         }
 
-        if (
-            this.computedValue === this.countryCode ||
-            (this.search !== null &&
-                Number.isNaN(
-                    parseInt(this.search.replace(TUI_MASK_SYMBOLS_REGEXP, ''), 10),
-                ))
-        ) {
-            this.updateSearch('');
+        if (!active && !this.allowText && this.nativeFocusableElement) {
+            this.nativeValue = this.nativeValue.replace(/-$/, '');
         }
-    }
 
-    onBackspace(event: Event): void {
-        const target = event.target as HTMLInputElement;
-
-        if (
-            (target.selectionStart || 0) <= this.nonRemovableLength &&
-            target.selectionStart === target.selectionEnd
-        ) {
-            event.preventDefault();
+        if (this.nativeValue === this.nonRemovablePrefix || this.isTextValue) {
```

Comment:
Why are we clearing text value?

Reply:
It is already existing behaviour for `main` branch:
https://taiga-ui.dev/components/input-phone#autocomplete


https://github.com/Tinkoff/taiga-ui/assets/35179038/5354ab7b-3b1f-4c46-b012-bc80d6ae232d

Reply:
I think that it is kind of feature 🤔

Reply:
I guess it is so only valid "named" phone numbers are there in the control.

Reply:
I guess it is so only valid "named" phone numbers are there in the control.